test/counter_server_example/src/counter_server_supervisor.erl

```diff
@@ -0,0 +1,73 @@
+%%% @doc
+%%%
+%%% Supervisor for the counter server. Responsible for maintaining the
+%%% permanent ETS table that the counter server uses to store state.
+%%%
+%%% @end
+%%%-------------------------------------------------------------------
+-module(counter_server_supervisor).
+
+%% API
+-export([ start_link/0
+        , start_link_no_vnet/0
+        ]).
+
+-behaviour(supervisor).
+-export([init/1]).
+
+%%%-------------------------------------------------------------------
+
+-include("counter_server.hrl").
+
+-define(SUPERVISOR, ?MODULE).
+
+%%%===================================================================
+%%% API functions
+%%%===================================================================
+
+start_link() ->
+  start_link(vnet).
+
+start_link_no_vnet() ->
+  start_link(no_vnet).
+
+%%%-------------------------------------------------------------------
+
+start_link(Arg) ->
+  SupervisorName =
+    case Arg of
+      no_vnet -> {local, ?SUPERVISOR};
+      vnet -> {via, vnet, ?SUPERVISOR}
+    end,
+  supervisor:start_link(SupervisorName, ?MODULE, [Arg]).
+
+%%%===================================================================
+%%% Supervisor callbacks
+%%%===================================================================
+
+init([Arg]) ->
+
+  ?TABLE = ets:new(?TABLE, [named_table, public]),
```

Comment:
The table name shall also be vnode-local in case of using `vnet`! Something like:

```erlang
TableName =
  case Arg of
    no_vnet -> ?TABLE;
    vnet -> vnet:tab(?TABLE)
  end,
```